ray-operator/controllers/ray/utils/validation.go

```diff
@@ -98,6 +106,10 @@ func ValidateRayJobStatus(rayJob *rayv1.RayJob) error {
 }
 
 func ValidateRayJobSpec(rayJob *rayv1.RayJob) error {
+	if len(rayJob.Name) > MaxRayJobNameLength {
```

Comment:
maybe we can (1) separate the logic into `ValidateRayJobMetadata` /  `ValidateRayServiceMetadata` (2) only pass CR spec to the function `ValidateXXXXSpec` as a follow up PR. This will be a good first issue.

Reply:
done. I can open the issue for (2) after this is merged.